assets/queries/crossplane/aws/elb_using_weak_ciphers/query.rego

```diff
@@ -0,0 +1,112 @@
+
+	result := {
+		"documentId": input.document[i].id,
+		"resourceType": resource.kind,
+		"resourceName": resource.metadata.name,
+		"searchKey": "spec.forProvider",
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": "sslPolicy should be defined with a secure protocol or cipher",
+		"keyActualValue": "sslPolicy is not defined",
+		"searchLine": common_lib.build_search_line(["spec", "forProvider"], []),
+	}
+}
+
+CxPolicy[result] {
+	resource := input.document[i]
+	startswith(resource.apiVersion, "elbv2.aws.crossplane.io")
+	resource.kind == "Listener"
+	forProvider := resource.spec.forProvider
+
+	policy:= forProvider.sslPolicy
+	weakCipher(policy)
+
+	result := {
+		"documentId": input.document[i].id,
+		"resourceType": resource.kind,
+		"resourceName": resource.metadata.name,
+		"searchKey": "spec.forProvider.sslPolicy",
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "sslPolicy should use a secure protocol or cipher",
+		"keyActualValue": "sslPolicy is using a weak cipher",
+		"searchLine": common_lib.build_search_line(["spec", "forProvider"], ["sslPolicy"]),
+	}
+}
+
+CxPolicy[result] {
+	resource := input.document[i]
+	resource.kind == "Composition"
+	resourceList := resource.spec.resources
+
+	startswith(resourceList[j].base.apiVersion, "elbv2.aws.crossplane.io")
+	resourceList[j].base.kind == "Listener"
+	forProvider := resourceList[j].base.spec.forProvider
+	not common_lib.valid_key(forProvider, "sslPolicy")
+
+	result := {
+		"documentId": input.document[i].id,
+		"resourceType": resourceList[j].base.kind,
+		"resourceName": resourceList[j].base.metadata.name,
+		"searchKey": sprintf("spec.resources.base.metadata.name={{%s}}}.spec.forProvider", [resourceList[j].base.metadata.name]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": "sslPolicy should be defined with a secure protocol or cipher",
+		"keyActualValue": "sslPolicy is not defined",
+		"searchLine": common_lib.build_search_line(["spec", "resources", j, "base", "spec", "forProvider"], []),
+	}
+}
+
+CxPolicy[result] {
+	resource := input.document[i]
+	resource.kind == "Composition"
+	resourceList := resource.spec.resources
+
+	startswith(resourceList[j].base.apiVersion, "elbv2.aws.crossplane.io")
+	resourceList[j].base.kind == "Listener"
+	forProvider := resourceList[j].base.spec.forProvider
+	
+	policy:= forProvider.sslPolicy
+	weakCipher(policy)
+
+	result := {
+		"documentId": input.document[i].id,
+		"resourceType": resourceList[j].base.kind,
+		"resourceName": resourceList[j].base.metadata.name,
+		"searchKey": sprintf("spec.resources.base.metadata.name={{%s}}}.spec.forProvider", [resourceList[j].base.metadata.name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "sslPolicy should use a secure protocol or cipher",
+		"keyActualValue": "sslPolicy is using a weak cipher",
+		"searchLine": common_lib.build_search_line(["spec", "resources", j, "base", "spec", "forProvider"], ["sslPolicy"]),
+	}
+}
+
+# IANA
+weakCipher(aux) {
```

Comment:
I suggest saving the weak ciphers in the common lib and use them in all the relative queries